backend/api/Database/Models/Mission.cs

```diff
@@ -103,6 +104,27 @@ public static MissionStatus MissionStatusFromString(string status)
                   )
             };
         }
+
+        public void CalculateEstimatedDuration()
+        {
+            decimal robotVelocity = (decimal)1.5 * 1000 / 60; // km/t => m/min
+            decimal efficiencyFactor = (decimal)0.20;
+            decimal inspectionTime = 2; // min/tag
+            int assumedXYMetersFromFirst = 20;
+
+            double distance = 0;
+            int numberOfTags = 0;
+            var prevPosition = new Position(PlannedTasks.First().Pose.Position.X + assumedXYMetersFromFirst, PlannedTasks.First().Pose.Position.Y + assumedXYMetersFromFirst, PlannedTasks.First().Pose.Position.Z);
+            foreach (var plannedTask in PlannedTasks)
+            {
+                numberOfTags += plannedTask.Inspections.Count;
+                var currentPosition = plannedTask.Pose.Position;
+                distance += Math.Abs(currentPosition.X - prevPosition.X) + Math.Abs(currentPosition.Y - prevPosition.Y);
```

Comment:
Should we use vector distance instead of euclidean distance here?

Reply:
It's rarely possible to drive diagonal so I think vector distance will be more representative for the driving pattern

Reply:
Good point